core/modules/wbase/MsgProcessor.h

```diff
@@ -38,22 +37,23 @@ namespace proto {
 }
 namespace wbase {
     class SendChannel;
+    class Task;
 }}} // End of forward declarations
 
 namespace lsst {
 namespace qserv {
 namespace wbase {
 
-/// MsgProcessor implementations handle incoming TaskMsg objects and write their
-/// results over a SendChannel
-class MsgProcessor
-    : public util::BinaryCallable<std::shared_ptr<util::VoidCallable<void> >,
-                                  std::shared_ptr<proto::TaskMsg>,
-                                  std::shared_ptr<SendChannel> > {
-public:
-    /// @return a cancellation function.
-    /// This allows the caller to request work stoppage.
-    virtual R operator()(A1 taskMsg, A2 replyChannel) = 0;
+/** MsgProcessor implementations handle incoming TaskMsg objects by creating a Task to write their
+ * results over a SendChannel
+ */
+struct MsgProcessor {
```

Comment:
It's now the parent class to Foreman, and it's a direct call to Forman::processMsg instead of creating a functor to do the same thing.